samples/kafka_confluent/receiver-assign/main.go

```diff
@@ -0,0 +1,60 @@
+/*
+ Copyright 2021 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	kafka_confluent "github.com/cloudevents/sdk-go/protocol/kafka_confluent/v2"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/client"
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+)
+
+var topic = "test-confluent-topic"
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	receiver, err := kafka_confluent.New(kafka_confluent.WithConfigMap(&kafka.ConfigMap{
+		"bootstrap.servers": "127.0.0.1:9092",
+		"group.id":          "test-confluent-offset-id",
+		// "auto.offset.reset":  "earliest",
+		"enable.auto.commit": "true",
+	}), kafka_confluent.WithReceiverTopics([]string{topic}))
+	if err != nil {
+		log.Fatalf("failed to create kafka protocol, %v", err)
+	}
+
+	defer receiver.Close(ctx)
+
+	c, err := cloudevents.NewClient(receiver, client.WithPollGoroutines(1))
```

Comment:
question: reason you set goroutines to `1`?

Reply:
I think that using multiple goroutines may break the order of messages in the Kafka partition, please correct me if I am wrong.

Reply:
Hm, ideally you'd have `WithPollGoroutines == kafka_partitions`, right?

Reply:
Emm I don't think so. It's a little bit complicated.

Cause Kafka uses multiple goroutines(consumer group) to poll messages from the partitions on the broker. 
Ideally, the `max number of the consumer instance in the group` = the `number of the Kafka topic partitions`, and each of consumer will get a unique message from that assigned partition. This allows Kafka consumption throughput to reach its maximum while ensuring local ordering (partition). All these features all provided by Kafka itself. 

From my current understanding, Cloudevents also uses a multi-goroutines model to improve receiver consumption speed. However, if multiple `PollGoroutines` are used in conjunction with Kafka's consumer group feature, it will disrupt the intended purpose of the Kafka consumer group model because we cannot guarantee message orderliness under the message key partitioning mechanism.

So I set the `WithPollGoroutines` to `1` based on the above consideration

Reply:
Just to check: you're saying the Confluent implementation handles goroutines per consumer group accordingly by default?

Reply:
Yes, by default, the consumer group associated with the confluent consumer manages all consumers, whether they are implemented as multiple goroutines or multiple threads.